### DISCUSSION TITLE
How to update i3?

### DISCUSSION THREAD

**Discussion by Sourish-Sharma** [i3/i3#6318] [Q&A] [answered]
I installed i3 through apt, and unfortunately its version 4.20.1 which does not support i3 gaps. I have a decent amount of configs, should I just reinstall it or is there any better method? I assume my old configs wont cause an issue right?

**Comment by okraits** [comment]
You can either:
- update your linux install to a newer version of your distro if that's possible
- use the Debian or Ubuntu repositories provided by the i3 project: https://i3wm.org/docs/repositories.html

You can keep your configs. If you are using some deprecated setting, i3 will tell you.

**Reply by Sourish-Sharma** [reply]
I see, if possible may I know what behaviour will happen if I just install a newer version of i3 while having the older install? or should I back up the .config, remove i3 then do a new install? reinstalling distro will not be worth atm.

**Reply by okraits** [reply, marked as answer]
If you update the i3 package with apt, the old version of i3 will be replaced with the new version.
You can keep your configs, the new version of i3 will behave like the old version.
